parsers/GT.py

```diff
@@ -1,163 +1,206 @@
+    results: dict[datetime, dict] = collections.defaultdict(dict)
+    for local_day, daily_payload in daily_payloads.items():
+        for row in daily_payload:
+            # The API returns hours in the range [1, 24], so each one refers to the
+            # past hour (e.g., 1 is the time period [00:00, 01:00)). Shift the hour
+            # so each index represents the hour ahead and is in the range [0, 24),
+            # e.g., hour 0 represents the period [00:00, 01:00).
+            dt_local = local_day + timedelta(hours=int(row["hora"]) - 1)
+            dt_utc = dt_local.astimezone(timezone.utc)
+
+            # ignore data outside of range of interest
+            if dt_utc < target_day_utc_start or dt_utc > target_day_utc_end:
+                continue
+
+            # For live data, the current hour's data isn't finished updating until the hour has passed,
+            # so the current hour should not be included in the results.
+            if dt_utc >= hour_now_utc:
+                continue
+
+            results[dt_utc][row["tipo"]] = row["potencia"]
+
+    return results
+
+
+@refetch_frequency(timedelta(days=1))
 def fetch_consumption(
-    zone_key: str = DEFAULT_ZONE_KEY,
+    zone_key: ZoneKey = DEFAULT_ZONE_KEY,
     session: Session | None = None,
     target_datetime: datetime | None = None,
     logger: Logger = getLogger(__name__),
-):
-    """Fetch a list of hourly consumption data, in MW, for the day of the
-    requested date-time.
-    """
+) -> list[dict]:
+    """Fetch a list of hourly consumption data, in MW, for the UTC day of the requested date-time."""
 
+    session = session or Session()
     target_datetime = (
-        datetime.now(TIMEZONE)
+        datetime.now(timezone.utc)
         if target_datetime is None
-        else target_datetime.astimezone(TIMEZONE)
+        else target_datetime.astimezone(timezone.utc)
+    )
+    api_data = _get_api_data(
+        session,
+        target_datetime=target_datetime,
+        target=ApiKind.CONSUMPTION,
+        num_backlog_days=0,
     )
 
-    api_data = _get_api_data(session, URL, target_datetime, target=ApiKind.CONSUMPTION)
-
-    day_datetime = datetime.combine(
-        target_datetime, time(), tzinfo=TIMEZONE
-    )  # truncate to day
-    results = [
-        {
-            "consumption": row["DEM SNI"],
-            "datetime": day_datetime + timedelta(hours=hour),
-            "source": DOMAIN,
-            "zoneKey": zone_key,
-        }
-        for hour, row in enumerate(api_data)
-    ]
-    return results
+    consumption_list = TotalConsumptionList(logger=logger)
+    for dt, row in api_data.items():
+        consumption = row["DEM SNI"]
+        if not consumption:
+            continue
+
+        consumption_list.append(
+            zoneKey=zone_key,
+            datetime=dt,
+            consumption=consumption,
+            source=SOURCE,
+        )
+    return consumption_list.to_list()
 
 
+@refetch_frequency(timedelta(days=1))
```

Comment:
note: I have set it to `1` but happy to increase this if it's preferred

The data API only supports asking for one day at a time so asking for e.g. two days of data every two days in reality results in the same number of HTTP calls being made than asking for one day of data every day. Happy to go with whichever combination of refetch frequency / num_backlog_days you prefer